extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -160,6 +161,10 @@ export function getElement(selector, el, selectionMethod) {
     return closestByTag(el, selector);
   } else if (selectionMethod == 'scope') {
     return el.parentElement.querySelector(selector);
+  } else if (selectionMethod == 'host') {
```

Comment:
So, I've looked around some more. I still think it's very conventional to support support `sectionMethod == 'host' || selector = ':host' || selector = ':root'`.

But the bigger question I have, from embed's point of view, what does this trigger mean:
```
"triggers": {
  "totalVisibility": {
    "on": "visible",
    "request": "visibility",
  }
}
```

Would we actually allow an embed to get the analytics event for page visibility? It seems misleading, since we never impress embeds when page is NOT visible.

I think in the scope of the embed's document, the most logical meaning of this is that the embed wants to known its own visibility. Thus, I believe that `!visibilitySpec` and `selectorMethod = host || selector = :host` are actually two different ways to ask for page-level visibility event and it's simply that interpretation of this is different whether the `amp-analytics` is in the main document or embed. This is not at all far-fetched - this is basically the same interpretation as we'd have with `window.onload` and `iframe.contentWindow.onload`.

So, the actual code structure I see:
```
if (!visibilitySpec || visibilitySpec.selectorMethod = 'host' || ...:host) {
  // This is either page-level visibility or embed visibility.
  if (isEmbed(contextElement)) {
    ...
  } else {
    viewerFor(...).whenVisible().then(() => {
      trigger();
    });
  }
} else {
  // This is element visibility - no change from current implementation.
}
```